src/clusterfuzz/_internal/bot/tasks/task_types.py

```diff
@@ -0,0 +1,161 @@
+
+  def execute(self, task_argument, job_type, uworker_env):
+    """Executes a utask locally."""
+    preprocess_result = utasks.tworker_preprocess(self.module, task_argument,
+                                                  job_type, uworker_env)
+
+    if preprocess_result is None:
+      return
+
+    input_download_url, _ = preprocess_result
+    utasks.uworker_main(input_download_url)
+    logs.log('Utask: done with preprocess and main.')
+
+
+class UTaskLocalExecutor(BaseTask):
+  """Represents an untrusted task. Executes it entirely locally and in
+  memory."""
+
+  def execute(self, task_argument, job_type, uworker_env):
+    """Executes a utask locally in-memory."""
+    uworker_input = utasks.tworker_preprocess_no_io(self.module, task_argument,
+                                                    job_type, uworker_env)
+    if uworker_input is None:
+      return
+    uworker_output = utasks.uworker_main_no_io(self.module, uworker_input)
+    if uworker_output is None:
+      return
+    utasks.tworker_postprocess_no_io(self.module, uworker_output, uworker_input)
+    logs.log('Utask local: done.')
+
+
+class PostprocessTask(BaseTask):
+  """Represents postprocessing of an untrusted task."""
+
+  def __init__(self, module):
+    del module
+    # We don't need a module, postprocess isn't a real task, it's one part of
+    # many different tasks.
+    super().__init__('none')
+
+  def execute(self, task_argument, job_type, uworker_env):
+    """Executes postprocessing of a utask."""
+    # These values are None for now.
+    del job_type
+    del uworker_env
+    input_path = task_argument
+    utasks.tworker_postprocess(input_path)
+
+
+class UworkerMainTask(BaseTask):
+  """Represents uworker main of an untrusted task. This should only be used for
+  tasks that cannot use Google Cloud batch (e.g. Mac)."""
+
+  # TODO(metzman): Merge with PostprocessTask.
+  def __init__(self, module):
+    # We don't need a module, uworker_main isn't a real task, it's one part of
+    # many different tasks.
+    del module
+    super().__init__('none')
+
+  def execute(self, task_argument, job_type, uworker_env):
+    """Executes uworker_main of a utask."""
+    # These values are None for now.
+    del job_type
+    del uworker_env
+    input_path = task_argument
+    utasks.uworker_main(input_path)
+
+
+COMMAND_TYPES = {
+    # TODO(metzman): Change analyze task away from in-memory.
+    'analyze': UTaskLocalExecutor,
+    'blame': TrustedTask,
+    'corpus_pruning': UTaskLocalExecutor,
+    'fuzz': UTaskLocalExecutor,
+    'impact': TrustedTask,
+    'minimize': UTaskLocalExecutor,
+    'progression': UTaskLocalExecutor,
+    'regression': UTaskLocalExecutor,
+    'symbolize': TrustedTask,
+    'unpack': TrustedTask,
+    'uworker_postprocess': PostprocessTask,
+    'upload_reports': TrustedTask,
+    'uworker_main': UworkerMainTask,
+    'variant': UTaskLocalExecutor,
+}
+
+
+def is_trusted_portion_of_utask(task_name):
```

Comment:
Can you add docstrings? I'm also a little unclear on what this is doing exactly.